routes/foodtrucks.go

```diff
@@ -149,3 +164,107 @@ func GetFoodTrucksHandler(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 	json.NewEncoder(w).Encode(foodTrucks)
 }
+
+func PutFoodTrucksHandler(w http.ResponseWriter, r *http.Request) {
+
+	// Checks for food truck ID
+	params := mux.Vars(r)
+	foodTruckID, foodTruckIDExists := params["foodTruckID"]
+	if !foodTruckIDExists {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	// Get user from context
+	_, userLoggedIn := r.Context().Value(middleware.UserKey).(string)
+
+	// Check for a user
+	if !userLoggedIn {
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+
+	foodTruckDecoder := json.NewDecoder(r.Body)
+	foodTruckDecoder.DisallowUnknownFields()
+
+	// Decode request
+	var currentFoodTruck updateFoodTruckRequest
+	err := foodTruckDecoder.Decode(&currentFoodTruck)
+	if err != nil {
+		log.Printf("ERROR: %v", err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	// Set tags to an empty array if they don't exist
+	tags := currentFoodTruck.Tags
+	if tags == nil {
+		tags = []string{}
+	}
+
+	// Determine which fields should be updated
+	var updateData bson.D
+
+	if currentFoodTruck.Name != "" {
+		updateData = append(updateData, bson.E{"name", currentFoodTruck.Name})
+	}
+	if currentFoodTruck.Address != "" {
+		updateData = append(updateData, bson.E{"address", currentFoodTruck.Address})
+	}
+	if len(currentFoodTruck.Location) != 0 {
```

Comment:
do you need to check for nil here since location is an array?

Reply:
i tried nil, it doesnt work. since location is an array, its checked by if theres nothing in the array